### PR TITLE
#18951 : When the root path ("/") is being requested for the System Hos…

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…t or any other Site, the API must return the internal `/System folder` instance. In specific circumstances, during the Reindex Process, the Folder Factory is failing to read the `/System folder` appropriately. By calling the `findSystemFolder()` method now, we're making sure that the API is either (1) retrieving de System Folder from cache or the DB, or (2) re-creating it in case it has been deleted somehow.

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

**Comment by dsilvam** [comment]
Approving but needs a test

**Review by fabrizzio-dotCMS — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Comment by nollymar** [comment]
Tests implemented

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/folders/business/FolderFactoryImplTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.folders.model.Folder;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FolderFactoryImplTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/folders/business/FolderFactoryImplTest.java:21]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5896947)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/folders/business/FolderFactoryImplTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package com.dotmarketing.portlets.folders.business;
+
+import static com.dotmarketing.portlets.folders.business.FolderAPI.SYSTEM_FOLDER_PARENT_PATH;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/folders/business/FolderFactoryImplTest.java:3]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.folders.business.FolderAPI.SYSTEM_FOLDER_PARENT_PATH'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5896947)

**Pull request merged**